crates/pairstat_nostd_internal/src/twopoint/unstructured.rs

```diff
@@ -150,52 +150,38 @@ impl<'a, R: Reducer, B: BinEdges> ReductionSpec for TwoPointUnstructured<'a, R,
         self.squared_distance_bin_edges.n_bins()
     }
 
-    #[inline(always)]
-    fn outer_team_loop_bounds(
-        &self,
-        _team_id: usize,
-        team_info: &StandardTeamParam,
-    ) -> (usize, usize) {
-        // require serial implementation TODO
-        assert_eq!(team_info.n_members_per_team, 1);
-        assert_eq!(team_info.n_teams, 1);
-
-        (0, self.points_a.n_points)
-    }
-
-    fn inner_team_loop_bounds(
-        &self,
-        outer_index: usize,
-        _team_id: usize,
-        team_info: &StandardTeamParam,
-    ) -> (usize, usize) {
-        // require serial implementation TODO
-        assert_eq!(team_info.n_members_per_team, 1);
-        assert_eq!(team_info.n_teams, 1);
-
-        if self.is_auto {
-            (outer_index + 1, self.points_a.n_points)
-        } else {
-            (0, self.points_b.n_points)
-        }
+    // we could actually eliminate this method if we really want to
+    fn team_loop_bounds(&self, team_id: usize, team_info: &StandardTeamParam) -> (usize, usize) {
+        segment_idx_bounds(self.points_a.n_points, team_id, team_info.n_teams)
```

Comment:
If `segment_idx_bounds` is only used here, I think it would be better to directly implement it here.